spec/constraints/authenticated_user_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe AuthenticatedUser do
```

Comment:
Sounds a plan, was thinking it should be extended to check permissions from a list first, pass in the required permission when the instance is created.